torch/csrc/distributed/c10d/NCCLUtils.hpp

```diff
@@ -923,6 +1001,44 @@ struct NCCLTraceBuffer {
     return pg_config;
   }
 
+  const std::map<std::string, std::map<std::string, std::string>>
+  getPgConfigJson() {
+    std::map<std::string, std::map<std::string, std::string>> result;
+    for (const auto& [pg_name, ranks] : pg_name_to_ranks_) {
+      auto pg_info = std::map<std::string, std::string>();
+      pg_info["name"] = std::get<0>(pg_name);
+      pg_info["desc"] = std::get<1>(pg_name);
+      pg_info["ranks"] = ranks_str(ranks);
+      result.emplace(std::get<0>(pg_name), pg_info);
+    }
+    return result;
+  }
+
+  std::string dump_json(
+      const std::optional<std::unordered_map<
+          std::string,
+          std::unordered_map<std::string, std::string>>>& ncclDumpMap,
+      bool includeCollectives,
+      bool onlyActive) {
+    json result;
+    result[version_key_str] = version_val_str;
+    result[pg_config_key_str] = getPgConfigJson();
+
+    // collective trace
+    if (includeCollectives) {
+      auto entries = getCollectiveTraceJson(onlyActive);
+      if (entries.size() > 0) {
+        result[entries_key_str] = entries;
+      }
+    }
+
+    if (ncclDumpMap.has_value()) {
+      result[nccl_comm_key_str] = ncclDumpMap.value();
```

Comment:
is ncclDumpMap.value() another json encoded string? does it encode nicely in a nested way this way? (does the user have to do a separate decode on this value after decoding the parent?)

Reply:
Yes, ncclDumpMap is a map of a map
```
std::unordered_map<
      std::string /* ncclUniqueID */,
      std::unordered_map<std::string, std::string> 
 ```
This becomes JSON rather nicely.